modules/core/src/lib/effect-manager.ts

```diff
@@ -4,7 +4,6 @@ import MaskEffect from '../effects/mask/mask-effect';
 import type Effect from './effect';
 
 const DEFAULT_LIGHTING_EFFECT = new LightingEffect();
```

Comment:
Maybe as a good practice, do the same for all effects? global state is generally better avoided and the gains from sharing globally must be microscopic...

Reply:
The default LightingEffect doesn't include directional lights and thus shadow mapping isn't used. Thus the `gl` context isn't required in this case

Reply:
OK sure not a big deal but arguing that some effects should be kept global because we happen to know that their implementation details permit it seems tenuous. Your PR shone a light on the fact that some effects need to be maintained per-context and the current global list won't work. No matter, I can understand the desire to minimize additional changes at the end of long complex PR, and this is easy enough to address down the road.